pkg/ccl/storageccl/export.go

```diff
@@ -91,29 +91,24 @@ func evalExport(
 	// TODO(dan): Move all this iteration into cpp to avoid the cgo calls.
 	// TODO(dan): Consider checking ctx periodically during the MVCCIterate call.
 	var entries int64
-	for {
-		entries = 0
-		iter := engineccl.NewMVCCIncrementalIterator(batch)
-		defer iter.Close()
-		iter.Reset(args.Key, args.EndKey, args.StartTime, h.Timestamp)
-		for ; iter.Valid(); iter.Next() {
-			key, value := iter.Key(), iter.Value()
-			if log.V(3) {
-				log.Infof(ctx, "Export %+v %+v", key, value)
-			}
-			entries++
-			if err := sst.Add(engine.MVCCKeyValue{Key: key, Value: value}); err != nil {
-				return storage.EvalResult{}, errors.Wrapf(err, "adding key %s", key)
-			}
+	iter := engineccl.NewMVCCIncrementalIterator(batch)
+	defer iter.Close()
+	iter.Reset(args.Key, args.EndKey, args.StartTime, h.Timestamp)
+	for ; iter.Valid(); iter.Next() {
+		key, value := iter.Key(), iter.Value()
+		if log.V(3) {
+			log.Infof(ctx, "Export %+v %+v", key, value)
 		}
-		err := iter.Error()
-		if _, ok := err.(*roachpb.WriteIntentError); ok {
-			continue
+		entries++
+		if err := sst.Add(engine.MVCCKeyValue{Key: key, Value: value}); err != nil {
+			return storage.EvalResult{}, errors.Wrapf(err, "adding key %s", key)
 		}
-		if err != nil {
-			return storage.EvalResult{}, err
+	}
+	if err := iter.Error(); err != nil {
+		if _, ok := err.(*roachpb.WriteIntentError); ok {
+			log.Warningf(ctx, "retrying export because of an unresolved intent: %s", err)
```

Comment:
It's a little weird to say that we're retrying in code that doesn't actually control the retrying -- the logic that calls this could easily be changed without this log message ever being updated.

Is this log message even really needed for anything other than your debugging of the test?